### PR TITLE
Shared DatePicker

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#6803

*Description of changes:*

* Added Shared DatePicker component.
* Applied design updates.
* Rebuilt Storybook, along with DatePicker Stories.

### CODE REVIEW

File: src/components/DatePicker/DatePicker.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            :nudge-bottom="nudgeBottom"
+            :nudge-left="nudgeLeft"
+            :nudge-right="nudgeRight"
+            transition="scale-transition"
+            offset-y
+            bottom
+            min-width="290"
+    >
+      <template v-slot:activator="{ on }">
+        <v-text-field id="date-text-field"
```
Comment (severinbeauvais) [inline src/components/DatePicker/DatePicker.vue:16]: Will you make the text field editable?

File: src/components/DatePicker/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@bcrs-shared-components/date-picker",
+  "version": "0.0.0",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@bcrs-shared-components/enums": "^1.0.10",
+    "@bcrs-shared-components/interfaces": "^1.0.16",
```
Comment (severinbeauvais) [inline src/components/DatePicker/package.json:9]: This component doesn't depend on enums or interfaces right?

File: src/components/DatePicker/DatePicker.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/DatePicker/DatePicker.vue:16]: I asked Tracey this when making it, and she said no at the time. 
Is this a different requirement for FED?

File: src/components/DatePicker/package.json (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/DatePicker/package.json:9]: It does not! Sorry, copypasta error.
Will remove until required.

File: src/components/DatePicker/DatePicker.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/DatePicker/DatePicker.vue:16]: I don't know either way, so "no" until it's "yes".

File: src/components/DatePicker/DatePicker.vue [hunk 3]
```diff
@@ -29,11 +29,11 @@
...
         />
       </template>
-      <v-date-picker width="490" v-model="dateText" :min="minDate" :max="maxDate">
+      <v-date-picker id="date-picker-calendar" width="490" v-model="dateText" :min="minDate" :max="maxDate">
         <template v-slot:default>
           <div>
-            <v-btn text color="primary" @click="emitDate(dateText)"><strong>OK</strong></v-btn>
-            <v-btn text color="primary" @click="emitCancel()">Cancel</v-btn>
+            <v-btn id="btn-done" text color="primary" @click="emitDate(dateText)"><strong>OK</strong></v-btn>
+            <v-btn id="btn-cancel" text color="primary" @click="emitCancel()">Cancel</v-btn>
```
Comment (severinbeauvais) [inline src/components/DatePicker/DatePicker.vue:36]: Is there a chance of getting duplicate IDs on a page? I assume it's not possible to have multiple date pickers open at once, but are these elements rendered even if the dp isn't showing?

**Review by severinbeauvais — COMMENTED** [review]
Awesome tests!

File: src/components/DatePicker/DatePicker.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/DatePicker/DatePicker.vue:36]: I don't believe they are. In the tests i check the existence of the calendar aspect of the component before activated and it comes back as `false`, so it isn't question of displayed vs hidden, more like a `v-if` i think. 

Screen Shot 2021-03-31 at 10 56 29 AM

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**